internal/util/update/resource-merge.go

```diff
@@ -242,6 +240,17 @@ func getSubDirsAndNonKrmFiles(root string) (sets.String, sets.String, error) {
 	return dirs, files, nil
 }
 
+func isMatch(matchFilesGlob []string, path string) (bool, error) {
```

Comment:
Go doc comment would be helpful. Also this can be easily unit-tested so expecting some tests.

Reply:
I see that this implementation is more clear. But we are duplicating the logic in kyaml. The ideal way I would imagine is to refactor the method in kyaml to something like this instead of `shouldSkipFile` which is confusing and is confusing to be used elsewhere. But for now we I am fine with this.

Reply:
Added go docs. I don't think this little function itself needs tests. This is tested through the public `ReplaceNonKRMFiles` function.